Packages/Core/Runtime/Listeners/AtomBaseListener.cs

```diff
@@ -69,20 +83,36 @@ private void OnDisable()
         /// <param name="item">The Event type.</param>
         public void OnEventRaised(T item)
         {
-            _unityEventResponse?.Invoke(item);
-            for (int i = 0; _actionResponses != null && i < _actionResponses.Count; ++i)
+            bool shouldRespond = _operator == AtomConditionOperators.And ? true : false;
+
+            // Evaluate conditions and decide whether to respond or not
+            for (int i = 0; _conditions != null && i < _conditions.Count; ++i)
             {
-                var action = _actionResponses[i];
+                var condition = _conditions[i];
 
-                if (action == null) continue;
+                if(condition == null) continue; // TODO: Use IsUnassigned when available
 
-                if (action is AtomAction<T> actionWithParam)
-                {
-                    actionWithParam.Do(item);
-                }
-                else
+                if(_operator == AtomConditionOperators.And) shouldRespond &= condition.Call(item);
+                if(_operator == AtomConditionOperators.Or) shouldRespond |= condition.Call(item);
```

Comment:
early return would be possible

    if(_operator == AtomConditionOperators.And && !shouldRespond) break;
    if(_operator == AtomConditionOperators.Or  &&  shouldRespond) break;